src/managers/organization/authentication-by-peers.ts

```diff
@@ -0,0 +1,301 @@
+import {
+  findById as findOrganizationById,
+  getUserOrganizationLink,
+  getUsers,
+} from '../../repositories/organization/getters';
+import { isEmpty } from 'lodash';
+import { findById as findUserById } from '../../repositories/user';
+import {
+  NotFoundError,
+  UserAlreadyAskedForSponsorshipError,
+} from '../../errors';
+import { isEligibleToSponsorship } from '../../services/organization';
+import { sendMail } from '../../connectors/sendinblue';
+import { updateUserOrganizationLink } from '../../repositories/organization/setters';
+import { getOrganizationsByUserId } from './main';
+import {
+  createModeration,
+  findPendingModeration,
+} from '../../repositories/moderation';
+
+const { SUPPORT_EMAIL_ADDRESS = 'moncomptepro@beta.gouv.fr' } = process.env;
+
+export const authenticateByPeers = async (
+  link: UserOrganizationLink
+): Promise<{ hasBeenAuthenticated: boolean }> => {
+  const { organization_id, user_id, is_external } = link;
+  const organizationUsers = await getUsers(organization_id);
+  const user = organizationUsers.find(({ id }) => id === user_id);
+  const organization = await findOrganizationById(organization_id);
+
+  // The user should be in the organization already
+  if (isEmpty(user) || isEmpty(organization)) {
+    throw new NotFoundError();
+  }
+
+  if (isEligibleToSponsorship(organization)) {
+    return { hasBeenAuthenticated: false };
+  }
+
+  await notifyAllMembers(link);
+
+  return { hasBeenAuthenticated: true };
+};
+export const notifyAllMembers = async ({
+  organization_id,
+  user_id,
+  is_external,
+}: UserOrganizationLink) => {
+  const organizationUsers = await getUsers(organization_id);
+  const user = organizationUsers.find(({ id }) => id === user_id);
+  const organization = await findOrganizationById(organization_id);
+
+  // The user should be in the organization already
+  if (isEmpty(user) || isEmpty(organization)) {
+    throw new NotFoundError();
+  }
+
+  const { email, given_name, family_name } = user;
+  const { cached_libelle } = organization;
+
+  // Email organization members of the organization
+  const usersInOrganization = await getUsers(organization_id);
+  const otherInternalUsers = usersInOrganization.filter(
+    ({ email: e, is_external, authentication_by_peers_type }) =>
+      e !== email && !is_external && !!authentication_by_peers_type
+  );
+  if (otherInternalUsers.length > 0) {
+    const user_label =
+      !given_name && !family_name ? email : `${given_name} ${family_name}`;
+    await sendMail({
+      to: otherInternalUsers.map(({ email }) => email),
+      subject: 'Votre organisation sur MonComptePro',
+      template: 'join-organization',
+      params: { user_label, libelle: cached_libelle, email, is_external },
+    });
```

Comment:
**thought**: the `sendMail` function type is very lax and is prone to errors.

One might want to ensure that the required params are always passed to the right template

```ts
import sendMail from "@/mail"
// [...]
sendMail.joinOrganization({
  to: otherInternalUsers.map(({ email }) => email),
  subject: 'Votre organisation sur MonComptePro',
  params: { user_label, libelle: cached_libelle, email, is_external },
})
```
where the send mail is some like :

```ts
import { sendMail } from "@/connectors/sendinblue";
// [...]
export const joinOrganization = sendMail<'join-organization', {
  user_label: string;
  libelle: string | null;
  email: string;
  is_external: boolean;
}>;
```

and 

```ts

export const sendMail = async <
  T extends RemoteTemplateSlug | LocalTemplateSlug,
  P = any
>({
  to = [],
  cc = [],
  subject,
  template,
  params,
}: {
  to: string[];
  cc?: string[];
  subject: string;
  template: T;
  params: P;
}) => {
  // [...]
}
```